tasks/service_stop_amq_broker_instance.yml

```diff
@@ -20,18 +20,39 @@
   when:
     - amq_broker_service_start is defined and amq_broker_service_start == true
     - amq_broker_path_st.stat.exists and amq_broker_path_st.stat.isdir is defined
-  become: true
-  become_user: "{{ amq_broker_user | default(amq_broker_default_user) }}"
-  command: "{{ amq_broker_install_dest }}/{{ amq_broker_instance_dir }}/bin/artemis-service stop"
-  register: stop_result
+  block:
+    - name: Stop broker instance via service
+      become: true
+      become_user: "{{ amq_broker_user | default(amq_broker_default_user) }}"
+      command: "{{ amq_broker_install_dest }}/{{ amq_broker_instance_dir }}/bin/artemis-service stop"
+  rescue:
+    - name: Kill JBoss AMQ 7 Broker
+      include: kill_amq_broker_instance.yml
 
 - name: Stop JBoss AMQ 7 Broker default instance
-  when: amq_broker_service_start is defined and amq_broker_service_start == false
-  become: true
-  become_user: "{{ amq_broker_user | default(amq_broker_default_user) }}"
-  command: "{{ amq_broker_install_dest }}/{{ amq_broker_instance_dir }}/bin/artemis stop"
-  register: stop_result
+  when:
+    - amq_broker_service_start is defined and amq_broker_service_start == false
+    - amq_broker_path_st.stat.exists and amq_broker_path_st.stat.isdir is defined
+  block:
+    - name: Stop broker instance
+      become: true
+      become_user: "{{ amq_broker_user | default(amq_broker_default_user) }}"
+      command: "{{ amq_broker_install_dest }}/{{ amq_broker_instance_dir }}/bin/artemis stop"
+  rescue:
+    - name: Kill JBoss AMQ 7 Broker
+      include: kill_amq_broker_instance.yml
 
-- name: Kill JBoss AMQ 7 Broker
-  when: stop_result | failed and amq_broker_force_kill is defined and amq_broker_force_kill == true
-  include: kill_amq_broker_instance.yml
+- name: Make sure no Artemis/AMQ 7 broker is running on machine
+  when: amq_broker_force_kill is defined and amq_broker_force_kill == true
+  block:
+    - name: Check running artemis process
+      command: 'pgrep -f "java.*artemis"'
+      register: bad_artemis_pid
+      msg: "Bad artemis= {{ bad_artemis_pid }}"
+      failed_when: bad_artemis_pid.rc == 0
+  rescue:
+    - name: SIGKILL any artemis java process
+      become: true
+      become_user: root
+      command: 'pkill -9 -f "java.*artemis"'
```

Comment:
**NOT AN ISSUE**: I am wondering if we should include procps-ng as a dependency that is checked. Thoughts?

Reply:
I think this is very dangerous. It can kill broker instances not running on the same user (this is a problem for Maestro). 

1. I don't think it needs to become root here.
2. I think a safer command line, using shell module instead of command, would be something like: 

> pgrep -U $(id -u {{ amq_broker_user }} ) -f '.\*java.\*artemis.\*'

Reply:
Nope that won't work in some cases. For example if broker was started by root, or any other user, who is *not* current {{ amq_broker_user }}, it will always FAIL and whole deployment would be wrong. Thus I added there strong kill for all artemis processes. 
As another option I was thinking about adding another parts of artemis CMD for example "-Dartemis.home" or similar to distinguish clearly brokers.

Also, your pgrep ... command is same as pkill -i {{ amq_broker_user }} ... - so it can be simplified a lot.

Reply:
<mtoth> what if we would put there amq_broker_allow_kill_all bool?
<opiske> if it defaults to false, then yes

Reply:
Commited changes to reflect this behavior. Please see latest commit(s).